x/staking/querier/querier.go

```diff
@@ -232,7 +234,7 @@ func queryDelegatorValidators(ctx sdk.Context, cdc *codec.Codec, req abci.Reques
 
 	errRes := cdc.UnmarshalJSON(req.Data, &params)
 	if errRes != nil {
-		return []byte{}, sdk.ErrUnknownAddress("")
+		return []byte{}, sdk.ErrInternal(fmt.Sprintf("failed to parse params: %s", err))
```

Comment:
Maybe `sdk.ErrFailedParams` or something similar?

Reply:
Hmmm, I'm not sure that is totally a good idea as we could have a multitude of decoding failures and we don't want a unique failure type/function for reach. Maybe `ErrFailedDecode` but that can be for another PR.